### PR TITLE
updates score_forecasts() function to ouput more zoltar-style scores …

### LINKED ISSUE #80: fix quantile aggregation issue in score_forecasts()

### PR DESCRIPTION
**Pull request by nickreich**
…with one row per model-location-target-forecast_date, many columns for each coverage, abs_error and wis.

closes #80

### CODE REVIEW

**Comment by nickreich** [comment]
Based on the updated specs in #80 I think this score_forecasts() function works. However, there are a few places, noted in the code where we may need additional column validation.

**Comment by nickreich** [comment]
This snippet runs one simple set of scores to that checked out manually against previously recorded scores:

```
library(covidHubUtils)
library(tidyverse)
forecasts <- load_latest_forecasts(models=c("COVIDhub-ensemble"), 
  last_forecast_date = "2020-10-01", 
  forecast_date_window_size = 7, 
  locations = "US", 
  targets = paste(1:4, "wk ahead inc death"), 
  source = "zoltar")

truth <- load_truth("JHU", target_variable = "inc death", locations = "US")

tmp <- score_forecasts(forecasts, truth)
```

**Review by ydhuang28 — APPROVED** [review]
With this change each (model + location + target + timezero) combination gets a set of scores, correct? I ran the vignette just to make sure.

We can open more issues for improving it, e.g., adding more complex scores, passing parameters to `eval_forecasts()`, etc., down the road.

File: R/score_forecasts.R [hunk 1]
```diff
@@ -71,27 +74,24 @@ score_forecasts <- function(
...
+
+  scores <- tibble::tibble(scoringutils::eval_forecasts(data = joint_df, 
+    by = observation_cols,
+    summarise_by = c(observation_cols, "range")),
+    interval_score_arguments = list(weigh = TRUE)) %>%
+    tidyr::pivot_wider(id_cols = observation_cols,
+      names_from = c("range"), 
+      values_from = c("coverage", "interval_score", "aem")) %>%
+    ## before next line: do we need to check to ensure aem_0 is a valid column name
+    dplyr::rename(abs_error = aem_0) %>%
```
Comment (nikosbosse) [inline R/score_forecasts.R:91]: haven't tested it, but if existence of aem_0 is of concern, we could maybe also do something like 
`dplyr::mutate(abs_error = unique(dplyr::starts_with("aem_"))` 
(not sure that exactly works.) But aem should be the same number for all quantiles.

**Review by nikosbosse — APPROVED** [review]
looks good!

**Pull request merged**